server/src/main/java/org/elasticsearch/index/engine/TranslogLeafReader.java

```
@@ -0,0 +1,207 @@
+        this.operation = operation;
+    }
+    @Override
+    public CacheHelper getCoreCacheHelper() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Terms terms(String field) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public NumericDocValues getNumericDocValues(String field) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BinaryDocValues getBinaryDocValues(String field) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SortedDocValues getSortedDocValues(String field) {
+        // TODO this can be removed in 7.0 and upwards we don't support the parent field anymore
+        if (field.startsWith(ParentFieldMapper.NAME + "#") && operation.parent() != null) {
+            return new AbstractSortedDocValues() {
+                @Override
+                public int docID() {
+                    return 0;
+                }
+
+                private final BytesRef term = new BytesRef(operation.parent());
+                private int ord;
+                @Override
+                public boolean advanceExact(int docID) {
+                    if (docID != 0) {
+                        throw new IndexOutOfBoundsException("do such doc ID: " + docID);
+                    }
+                    ord = 0;
+                    return true;
+                }
+
+                @Override
+                public int ordValue() {
+                    return ord;
+                }
+
+                @Override
+                public BytesRef lookupOrd(int ord) {
+                    if (ord == 0) {
+                        return term;
+                    }
+                    return null;
+                }
+
+                @Override
+                public int getValueCount() {
+                    return 1;
+                }
+            };
+        }
+        return null;
```

should we throw an exception?